packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -125,15 +124,18 @@ export default function NewTemplate( {
 
 			// Set template before navigating away to avoid initial stale value.
 			setTemplate( newTemplate.id, newTemplate.slug );
-
 			// Switch to edit mode.
 			setCanvasMode( 'edit' );
 
 			// Navigate to the created template editor.
-			history.push( {
-				postId: newTemplate.id,
-				postType: newTemplate.type,
+			window.queueMicrotask( () => {
```

Comment:
Why we wrap this with `window.queueMicrotask`?

Reply:
Because we have two hooks synchronizing state with history and if we do them synchronously, one will override the other. I'm working on a follow-up that hopefully simplifies things and keep things also more backward compatible in terms of URLs.